interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -239,27 +217,31 @@ private class CatsEffect[R] extends CatsMonadError[R, Throwable] with effect.Asy
   override final def uncancelable[A](fa: RIO[R, A]): RIO[R, A] =
     fa.uninterruptible
 
-  override def guarantee[A](fa: RIO[R, A])(finalizer: RIO[R, Unit]): RIO[R, A] =
+  override final def guarantee[A](fa: RIO[R, A])(finalizer: RIO[R, Unit]): RIO[R, A] =
     fa.ensuring(finalizer.orDie)
+
+  override final def continual[A, B](fa: RIO[R, A])(f: Either[Throwable, A] => RIO[R, B]): RIO[R, B] =
+    ZIO.uninterruptibleMask(_(fa).either.flatMap(f))
```

Comment:
Yes, I think this is correct from inspecting the implementation of `Concurrent.continual`. I thought at first that `f` shouldn't run uninterruptibly, but in `Concurrent.continual` it is executed under `guaranteeCase` which (I assume) runs a finalizer uninterruptibly.